src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/abstractItems/AContainer.java

```diff
@@ -416,33 +416,39 @@ protected MachineRecipe findNextRecipe(BlockMenu inv) {
             }
         }
 
-        Map<Integer, Integer> found = new HashMap<>();
+        Map<MachineRecipe, Map<Integer, Integer>> matched = new HashMap<>();
 
         for (MachineRecipe recipe : recipes) {
+            Map<Integer, Integer> found = new HashMap<>();
+
             for (ItemStack input : recipe.getInput()) {
                 for (int slot : getInputSlots()) {
+                    if (found.containsKey(slot)) continue;
                     if (SlimefunUtils.isItemSimilar(inventory.get(slot), input, true)) {
                         found.put(slot, input.getAmount());
                         break;
                     }
                 }
-            }
-
-            if (found.size() == recipe.getInput().length) {
-                if (!InvUtils.fitAll(inv.toInventory(), recipe.getOutput(), getOutputSlots())) {
-                    return null;
-                }
-
-                for (Map.Entry<Integer, Integer> entry : found.entrySet()) {
-                    inv.consumeItem(entry.getKey(), entry.getValue());
+                if (found.size() == recipe.getInput().length) {
+                    matched.put(recipe, found);
                 }
-
-                return recipe;
-            } else {
-                found.clear();
             }
         }
+        if (matched.isEmpty()) return null;
+        Map.Entry<MachineRecipe, Map<Integer, Integer>> recipe = matched.entrySet().stream().max((x, y) -> {
+            int sizex = ItemUtils.getAllItemTypeAmount(x.getKey().getInput()) * 1000 + ItemUtils.getAllItemAmount(x.getKey().getInput());
+            int sizey = ItemUtils.getAllItemTypeAmount(y.getKey().getInput()) * 1000 + ItemUtils.getAllItemAmount(y.getKey().getInput());
+            return Integer.compare(sizex, sizey);
+        }).get();
```

Comment:
Please clean this up.
This is super hard to read.
Also what is x and what is y please give them good names